logic/pulsed/sequence_generator_logic.py

```diff
@@ -117,10 +118,19 @@ def __init__(self, config, **kwargs):
         # directory for additional generate methods to import
         # (other than logic.predefined_generate_methods)
         if 'additional_methods_dir' in config.keys():
-            if not os.path.exists(config['additional_methods_dir']):
+            if isinstance(config['additional_methods_dir'], str) and not os.path.exists(config['additional_methods_dir']):
```

Comment:
Instead of defining the same operation for the single path and the all paths in a for loop, I would simply wrap a single str path into a list of length 1 and always iterate over that.

Reply:
good point, I fixed it.